### ISSUE TITLE
check.ci.ocaml.org is not responding

### ISSUE THREAD

**Issue by kit-ty-kate** [ocaml/infrastructure#84]

**Comment by mtelvers** [comment]
Indeed, I looked in on it a few minutes ago, and the CPU is nailed at 100%.  I was hoping that it was a temporary condition.  I'll give it a minute more, and failing that, I'll restart it.

**Comment by kit-ty-kate** [comment]
It’s still inaccessible for me

**Comment by mtelvers** [comment]
I've turned on debugging output in a hope to get some more insight into it.

**Comment by mtelvers** [comment]
I have temporarily relocated the majority of the old logs, and now it's up and running and at full speed.  I'll continue to investigate.

**Comment by mtelvers** [comment]
@kit-ty-kate This appears to be due to the large number of historical logs.  The application doesn't crash; it is just very slow to respond.  Looking at the folder on the server, it looks as though the logs from May 2019 through to December 2020 have been previously _archived_.  How many logs do we need to keep available?  Is there value in old logs or just the latest N logs?  At the moment, I have the current month's logs plus the log from the first run in each month back to January 2021.  The performance seems ok.

**Comment by kit-ty-kate** [comment]
I’m not really a user of check.ci.ocaml.org anymore but I think @mseri and @raphael-proust could answer.
By default I’d say "keep as much as possible" in case there is a problem and you want to know what worked and when it started failing but that’s just me.

Unless it’s being spammed by bots, knowing the code I’m not sure why the number of archives would be the cause of a performance issue. The number of logs shouldn’t affect anything past the initialization phase.

**Comment by mseri** [comment]
I agree with Kate. The longer we can keep them the easier is for us to go back and check if there are sudden differences and what failures we had in the past
